### PR TITLE
Added DGESVX from lapack.

### PR DESCRIPTION
**Pull request by geirev**
**Issue**
Resolves #my_issue
I needed LU solver that returns condition number to check numerical stability.

**Approach**
Implemented interface to dgesvx (which is an extension of dgesv already in matrix_lapack.cpp)

**Pre un-WIP checklist**
- [ ] Have completed graphical integration test steps
Tested in real application and output is identical to dgesv. 
There are no built-in tests in libres for these routines.
No other ERT routines are currently calling dgesvx.

**Depends on**
* Statoil/libecl#

### CODE REVIEW

**Comment by ghost** [comment]
Can one of the admins verify this patch?

**Comment by markusdregi** [comment]
test this please

**Comment by markusdregi** [comment]
@geirev Could you please write a small test in `lib/res_util/tests/ert_util_matrix_lapack.cpp` demonstrating that the function works on a small case? Besides that, this is ready to be merged.

**Comment by geirev** [comment]
Ok, I will try :-)

**Comment by markusdregi** [comment]
test this please

**Comment by markusdregi** [comment]
Thanks for the updated version. When Jenkins also have tested, we can merge this.

File: lib/res_util/tests/ert_util_matrix_lapack.cpp [hunk 1]
```diff
@@ -82,11 +82,52 @@ void test_det2() {
...
+     matrix_random_init( m1 , rng );
+     matrix_assign(m2 , m1);
+
+     matrix_random_init( b1 , rng );
+     matrix_assign(b2 , b1);
+
+     matrix_dgesv( m1 , b1  );
+     matrix_dgesvx( m2 , b2, &rcond  );
+
+     test_assert_true( matrix_equal( b1 , b2 ) );
```
Comment (markusdregi) [inline lib/res_util/tests/ert_util_matrix_lapack.cpp]: This is the assertion failing the test. Does it pass locally?

**Comment by geirev** [comment]
With repeated runs, the original test failed some of the times. The problem is that there is no tolerance in the matrix_equal function (as there is in the python tests) and sometimes the results are different at the level of numerical truncation errors. Thus, the test should compare if || b1 - b2 || < epsilon where epsilon is a number related to the numerical precision and the condition number of the matrix that is inverted. I don't know how to get around this with the existing tests. However, I made a routine, matrix_similar that compares two matrices within a given tolerance, and I think this can also be useful later to test more complex numerical schemes where truncation errors make a difference. I added a test for matrix_similar as well in lib/res_util/tests/ert_util_matrix_lapack.cpp.

**Review by markusdregi — APPROVED** [review]
Very good! Thank you for your contribution.

**Pull request merged**